crates/sui-json-rpc/src/read_api.rs

```diff
@@ -324,7 +331,7 @@ impl ReadApiServer for ReadApi {
         let opts = opts.unwrap_or_default();
         let mut temp_response = IntermediateTransactionResponse::new(digest);
 
-        if opts.show_input {
+        if opts.show_input || opts.show_object_changes {
```

Comment:
Where do we use `temp_response.transaction` when `opts.show_input` is false but `opts.show_object_changes` is true? Might worth an inline comment

Reply:
the input is needed to retrieve the sender address.